src/Signup/components/Signup.js

```diff
@@ -0,0 +1,115 @@
+import React, { Component } from 'react';
+import { Image, View, Text, TouchableOpacity, TextInput } from 'react-native';
+import PropTypes from 'prop-types';
+
+import styles from '../components/Signup.style';
+
+const logoImg = require('../../globals/EventAllLightLong.png');
+
+class SignupScreen extends Component {
+  static propTypes = {
+    navigation: PropTypes.shape({
+      navigate: PropTypes.func.isRequired,
+    }).isRequired,
+  };
+
+  state = {
+    username: '',
+    password: '',
+    confirmedPassword: '',
+    passwordsMatch: false,
+  };
+
+  verifyFBSignup = () => {
+    // Facebook authentication here
+  }
+
+  confirmPassword = () => {
+    if (this.state.password === this.state.confirmedPassword) {
+      // remove error message
+      this.state.passwordsMatch = true;
```

Comment:
Oops I forgot to catch this: make this `this.setState({ passwordMatch: true })`
mutation state directly leads to undefined behavior. Sorry I didn't catch this earlier